devfiles/src/classes/widgets/TableWidget.tsx

```diff
@@ -130,6 +132,42 @@ export default class TableWidget extends Widget {
             );
         }
 
+        //Search bar
+        const searchBar = (
+            <span>
+                <span>Find Row:</span>
+                <input
+                    style={{ width: '50%', marginLeft: '5px', display: 'inline' }}
+                    className='form-control'
+                    type='text'
+                    defaultValue={this.searchState}
+                    list={
+                        this.tableEntries.length < this.searchMax
+                            ? this.uuid.toString() + '-search'
+                            : undefined
+                    }
+                    onChange={(event) => {
+                        this.bouncySearchState = event.currentTarget.value;
+                        const oldState = this.bouncySearchState;
+                        setTimeout(() => {
+                            if (oldState === this.bouncySearchState) {
+                                this.searchState = this.bouncySearchState;
```

Comment:
The timeout here is 300ms. Say the user presses on a 100ms, then presses on b in 100 ms, then presses on delete in 99ms so there's now just "a". Now there'll be three refreshes, more than needed?